src/ray/util/process.h

```diff
@@ -73,14 +74,18 @@ class Process {
   /// \param[in] decouple True iff the parent will not wait for the child to exit.
   /// \param[in] env Additional environment variables to be set on this process besides
   /// the environment variables of the parent process.
-  explicit Process(const char *argv[], void *io_service, std::error_code &ec,
-                   bool decouple = false, const ProcessEnvironment &env = {});
+  explicit Process(StartupToken token, const char *argv[], void *io_service,
```

Comment:
Instead of passing in a StartupToken, can we create one inside Process constructor? The global atomic counter can then be the class member variable.